tls/s2n_client_hello.c

```diff
@@ -286,15 +286,22 @@ int s2n_process_client_hello(struct s2n_connection *conn)
      * Negotiate protocol version, cipher suite, ALPN, select a cert, etc. */
     struct s2n_client_hello *client_hello = &conn->client_hello;
 
+    const struct s2n_cipher_preferences *cipher_preferences;
+    GUARD(s2n_connection_get_cipher_preferences(conn, &cipher_preferences));
+
+    /* Ensure that highest supported version is set correctly */
+    if (!s2n_cipher_preference_is_tls13_required(cipher_preferences)) {
```

Comment:
`s2n_cipher_preference_is_tls13_required` returns 1, 0 or -S2N_ERROR on error.  i suggest your use here checks it accordingly.

suggest to use S2N_SUCCESS to indicate successful match on that function.

Reply:
unfortunately using S2N_SUCCESS w/ S2N_FAILURE makes it hard to test using S2N_EXPECT_FAILURE() as the test runner expects an error to be set, the call uses 1 = true, 0 = false, -1 = error tho.